CMakeLists.txt

```diff
@@ -1,17 +1,76 @@
-cmake_minimum_required( VERSION 2.8 )
+cmake_minimum_required (VERSION 2.8)
 
-set(MODULES_INSTALL_PREFIX  "share/opm" CACHE STRING "The modules will be installed in: \${CMAKE_INSTALL_PREFIX}/\${MODULES_INSTALL_PREFIX}/cmake/Modules")
+set( OPM_COMMON_ROOT "" CACHE PATH "Root directory containing OPM related cmake modules")
+option(SIBLING_SEARCH "Search for other modules in sibling directories?" ON)
 
-install(DIRECTORY cmake/Modules DESTINATION ${CMAKE_INSTALL_PREFIX}/${MODULES_INSTALL_PREFIX}/cmake)
-install(DIRECTORY cmake/Scripts DESTINATION ${CMAKE_INSTALL_PREFIX}/${MODULES_INSTALL_PREFIX}/cmake)
-install(DIRECTORY cmake/Templates DESTINATION ${CMAKE_INSTALL_PREFIX}/${MODULES_INSTALL_PREFIX}/cmake)
+if(NOT OPM_COMMON_ROOT)
+   find_package(opm-common QUIET)
+endif()
 
-list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/Modules)
-include(OpmInit)
+if (opm-common_FOUND)
+   include(OpmInit)
+else()
+   if (NOT OPM_COMMON_ROOT AND SIBLING_SEARCH)
+      set(OPM_COMMON_ROOT ${PROJECT_SOURCE_DIR}/../opm-common)
```

Comment:
the CMakeLists.txt was copied from opm-core. I did not really go into semantic mode when adding it. ;)